src/plugins/formulas/formulas.js

```diff
@@ -285,6 +280,69 @@ export class Formulas extends BasePlugin {
     });
   }
 
+  /**
+   * Renders dependent sheets (handsontable instances) based on the changes - list of the
+   * recalculated dependent cells.
+   *
+   * @private
+   * @param {object[]} changedCells The values and location of applied changes within HF engine.
+   */
+  renderDependentSheets(changedCells) {
+    const affectedSheetIds = new Set();
+
+    changedCells.forEach((change) => {
+      // For the Named expression the address is empty, hence the `sheetId` is undefined.
+      const sheetId = change?.address?.sheet;
+
+      if (sheetId !== void 0) {
+        if (!affectedSheetIds.has(sheetId)) {
+          affectedSheetIds.add(sheetId);
+        }
+
+        if (sheetId === this.sheetId) {
+          const { row, col } = change.address;
+
+          // It will just re-render certain cell when necessary.
+          this.hot.validateCell(this.hot.getDataAtCell(row, col), this.hot.getCellMeta(row, col), () => {});
+        }
+      }
+    });
+
+    const hotInstances = new Map(
+      getRegisteredHotInstances(this.engine)
+        .map(hot => [hot.getPlugin('formulas').sheetId, hot])
+    );
+
+    affectedSheetIds.forEach((sheetId) => {
+      if (sheetId !== this.sheetId) {
+        const hot = hotInstances.get(sheetId);
+
+        hot.render();
+        hot.view?.adjustElementsSize();
```

Comment:
I don't like the tightly coupled instances. Previous approach was event driven and bounded to the current instance only

It's a working solution tho, so I will accept the PR

Reply:
I think both approaches are good enough. The strict event-driven approach probably would be possible to achieve but I don't get more time to work on this issue.

I agree that it's a working solution, there are some new tests so in the future this can be refactored without breaking change.